src/poliastro/contrib/extract_czml.py

```diff
@@ -0,0 +1,288 @@
+            The pixel offset (up and right)
+        color: list (int)
+            Rgba path color
+        width: int
+            Path width
+        show: bool
+            Indicates whether the path is visible
+        """
+        if pixel_offset is not None:
+            self.parse_dict_tuples(
+                [i, "label", "pixelOffset"], [("cartesian2", pixel_offset)]
+            )
+        if color is not None:
+            self.parse_dict_tuples(
+                [i, "path", "material", "solidColor", "color"], [("rgba", color)]
+            )
+        if width is not None:
+            self.parse_dict_tuples([i, "path"], [("width", width)])
+        if show is not None:
+            self.parse_dict_tuples([i, "path", "show"], [("boolean", show)])
+
+    def change_label_params(
+        self, i, fill_color=None, outline_color=None, font=None, text=None, show=None
+    ):
+        """
+        Change the label parameters.
+
+        Parameters
+        ----------
+        i : int
+            Referred body (count starts at 1)
+        fill_color: list (int)
+            Fill Color in rgba format
+        outline_color: list (int)
+            Outline Color in rgba format
+        font: str
+            Set label font style and size (CSS syntax)
+        text: str
+            Set label text
+        show: bool
+            Indicates whether the label is visible
+        """
+        if fill_color is not None:
+            self.parse_dict_tuples([i, "label", "fillColor"], [("rgba", fill_color)])
+        if outline_color is not None:
+            self.parse_dict_tuples(
+                [i, "label", "outlineColor"], [("rgba", outline_color)]
+            )
+        if font is not None:
+            self.parse_dict_tuples([i, "label"], [("font", font)])
+        if text is not None:
+            self.parse_dict_tuples([i, "label"], [("text", text)])
+        if show is not None:
+            self.parse_dict_tuples([i, "label"], [("show", show)])
+
+    def extract(self, ext_location=None):
+        """
+        Parameters
+        ----------
+        ext_location : str
+            Path to extract your file to, if not path is given, return the dump in the console
+        """
+
+        return json.dumps(list(self.czml.values()))
+
+    def add_orbit(self, orbit, N):
+        """
+        Adds an orbit
+
+        Parameters
+        ----------
+        orbit: poliastro.Orbit
+            Orbit to be added
+        N: int
+            Number of sample points
+        """
+
+        self.i += 1
+
+        if orbit.epoch < Time(self.start_epoch):
+            orbit = orbit.propagate(Time(self.start_epoch) - orbit.epoch)
+        elif orbit.epoch > Time(self.end_epoch):
+            raise ValueError(
+                "The orbit's epoch cannot exceed the constructors ending epoch"
+            )
+
+        self.orbits[self.i] = [orbit, N, orbit.period, orbit.epoch]
+        self.init_orbit_packet(self.i)
+
+    def del_orbit(self, i):
```

Comment:
I propose that we remove this method. If someone wants to "delete an orbit", they should create another extractor class. Maybe this makes the rest of the code simpler.

Reply:
I tend to agree, but I'm a bit reluctant to introduce this level of immutability, especially when the only alternative to defining one wrong orbit is starting all over again.

Reply:
It's true, but if you think about the workflow, most people will use editors like Spyder, Jupyter notebooks, Jupyter Lab... I don't think it's a big burden to change one or two lines of code, and from the implementation side it simplifies things a lot.